### ISSUE TITLE
AWS SDK Example

### ISSUE THREAD

**Issue by mgvarley** [apex/apex#520]
Would it be possible to have an example of using the aws-sdk with Apex?  I imagine that this is a very common use case for Lambda.  I am trying to get up and running with Apex using Node following the excellent babel-webpack example but struggling when it comes to querying a DynamoDB.  I have tried adapting the example using the async / await syntax with AWS.DynamoDB.DocumentClient but cannot seem to get any data to return.  A specific example for this would be hugely appreciated.

**Comment by tj** [comment]
Yeah that would be good to add! I don't use Node in Lambda personally, but if anyone wants to take a stab at this that would be awesome. I imagine it shouldn't really be any different than using the aws sdk outside of Lambda

**Comment by mgvarley** [comment]
Thanks for the quick reply tj, I managed to get an example running, was struggling with the aws-sdk and es7 async / await syntax, turns out aws-sdk now supports promises out of the box although documentation is a little cryptic!

Here is a simple example, a little hacky but does the job, it makes a call to DynamoDB to retrieve some data (payload) based upon a partition key (id) and sort key (dataset).  If anyone wants to improve on this for the Apex examples they are more than welcome.

`
import λ from 'apex.js'
import 'babel-polyfill'
import AWS from 'aws-sdk'

// Set the AWS region
AWS.config.update({
  region: 'eu-west-1'
})

// Constants
const DB_TABLE = '<MY_DB_TABLE>'

// Use the DynamoDB DocumentClient to simplify querying
var docClient = new AWS.DynamoDB.DocumentClient()

// Query the database
async function queryDb(id, dataset) {
  // Build DB request
  let params = {
    TableName : DB_TABLE,
    Key: {
      id: id,
      dataset: dataset
    }
  }
  // Query DynamoDB using promise and return result
  let data = await docClient.get(params).promise()
  return (data && data.Item && data.Item.payload) ? {
    id: id,
    dataset: dataset,
    data: data.Item.payload
  } : {
    err: `No data found in ${DB_TABLE} for ${id}`
  }
}

// Lambda function to query Dynamo DB database by id (partition key) and dataset (sort key)
export default λ(e => {
  console.log(`Querying ${e.dataset} for ${e.id}`)
  let res = queryDb(e.id, e.dataset)
  return res
})
`

Note that for this to work the role generated by Apex needs to be granted read permissions on the DynamoDB table in question.  This can be done by creating a new policy and attaching it to the Apex role.  Here is mine:

`
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Sid": "Stmt1470668095000",
            "Effect": "Allow",
            "Action": [
                "dynamodb:BatchGetItem",
                "dynamodb:GetItem",
                "dynamodb:Query",
                "dynamodb:Scan"
            ],
            "Resource": [
                "arn:aws:dynamodb:eu-west-1:<MY_AWS_ACCOUNT>:table/<MY_DB_TABLE>"
            ]
        }
    ]
}
`

Finally, thanks again for Apex, really takes the pain out of Lamdba development.

**Comment by andreineculau** [comment]
I guess this was meant for https://github.com/apex/node-apex ?

**Comment by jaysonesmith** [comment]
Is this still a valid issue for this repo?
